test/BilateralSwap/bilateralSwap.js

```diff
@@ -0,0 +1,152 @@
+/* global artifacts, expect, contract, beforeEach, it:true */
+// ### External Dependencies
+const BN = require('bn.js');
+const crypto = require('crypto');
+const { padLeft, sha3 } = require('web3-utils');
+
+
+// ### Internal Dependencies
+const exceptions = require('../exceptions');
+const { t2, GROUP_MODULUS } = require('../../aztec-crypto-js/params');
+const bilateralProof = require('../../aztec-crypto-js/proof/bilateralProof');
+const helpers = require('../../aztec-crypto-js/proof/helpers');
+
+// ### Artifacts
+const BilateralSwap = artifacts.require('../../contracts/AZTEC/BilateralSwap');
+const BilateralSwapInterface = artifacts.require('../../contracts/AZTEC/BilateralSwapInterface');
+
+const { toBytes32 } = require('../../aztec-crypto-js/utils/utils');
+
+BilateralSwap.abi = BilateralSwapInterface.abi;
+
+/*
+Outline of test:
+- This needs to be able to run on Truffle - to test the smart contract feature
+- Construct success states
+- Generate a proof using the BilateralSwap javascript proof generation code
+- Validate that it works, first using the javascript proof verification code
+- Validate that it works, using the bilateral swap smart construct
+*/
+
+contract('BilateralSwap tests', (accounts) => {
+    describe('success states', () => {
+        let BilateralSwap;
+        let testNotes;
+
+        beforeEach(async () => {
+            BilateralSwap = await BilateralSwap.new(accounts[0]);
+            const makerNoteValues = [10, 20];
+            const takerNoteValues = [10, 20];
+            testNotes = helpers.makeTestNotes(makerNoteValues, takerNoteValues);
+        });
+
+        it('validate that the Javascript proof is constructed correctly', () => {
+            const { proofData, challenge } = bilateralProof.constructBilateralSwap(testNotes, accounts[0]);
+            const result = bilateralProof.verifyBilateralSwap(proofData, challenge, accounts[0]);
+            expect(result).to.equal(true);
+        });
+
+        it('validate that the smart contract can verify the bilateral swap proof', async () => {
```

Comment:
we should add test cases for where the a bid/ask note pair has a value of '0', and validate that the proof passes